server/src/main/java/org/opensearch/Build.java

```diff
@@ -215,8 +214,8 @@ public static Build readBuild(StreamInput in) throws IOException {
         // TODO - clean this up when OSS flavor is removed in all of the code base
         // (Integ test zip still write OSS as distribution)
         // See issue: https://github.com/opendistro-for-elasticsearch/search/issues/159
-        if (in.getVersion().onOrAfter(LegacyESVersion.V_6_3_0)) {
-            flavor = in.readString();
+        if (in.getVersion().onOrAfter(LegacyESVersion.V_6_3_0) && in.getVersion().before(Version.V_1_3_0)) {
```

Comment:
Trying to understand, can we just simply use `Version.CURRENT`?

Reply:
BWC testing would fail... This is for reading off the wire from remote nodes between 6.3.0 and 1.3.0 excluded. (that covers all 1.2.x versions but 1.3.0 does not send flavor).

Reply:
THat makes sense. Thank you!